gps_handler.py

```diff
@@ -0,0 +1,97 @@
+import smbus
+import signal
+import sys
+import asyncio
+import typing
+import logging
+from ethernet_handler import ethernet_put
+
+i2c_bus = smbus.SMBus(1) # I2C bus for GPS Shield
+ADDRESS = 0x42 # GPS Shield I2C Address
+GPS_INTERVAL = 1 # Polling frequency in seconds
+
+def _connect_bus():
+    """Connect to the I2C bus. Function from ozzmaker.com"""
+    global i2c_bus
+    i2c_bus = smbus.SMBus(1)
+
+def _parse_response(gps_line) -> typing.Union[str, None]:
+    """Parse the GPS response."""
+
+    # Check #1, make sure '$' doesnt appear twice
+    if gps_line.count(36) != 1:
+        return None
+
+    # Check #2, 83 is maximum NMEA sentence length.
+    if len(gps_line) >= 84:
+        return None
+
+    # Check #3, Make sure that only readable ASCII charaters and Carriage Return are seen.
+    valid_chars = set(range(32,123)).union({13})
```

Comment:
You could use the string library here:
```import string``` way up where your imports are, and then:
```suggestion
    valid_chars = string.printable
```

Reply:
Honestly can prob just replace valid chars with string.printable wherever it's called (assuming this works)